internal/webhooks/runtime/extensionconfig_webhook.go

```diff
@@ -93,10 +101,121 @@ func (webhook *ExtensionConfig) validate(_ context.Context, _, _ *runtimev1.Exte
 			"can be set only if the RuntimeSDK feature flag is enabled",
 		)
 	}
+
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateExtensionConfigSpec(newExtensionConfig)...)
+
+	if len(allErrs) > 0 {
+		return apierrors.NewInvalid(runtimev1.GroupVersion.WithKind("ExtensionConfig").GroupKind(), newExtensionConfig.Name, allErrs)
+	}
 	return nil
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
 func (webhook *ExtensionConfig) ValidateDelete(_ context.Context, _ runtime.Object) error {
 	return nil
 }
+
+func validateExtensionConfigSpec(e *runtimev1.ExtensionConfig) field.ErrorList {
+	var allErrs field.ErrorList
+
+	specPath := field.NewPath("spec")
+
+	if e.Spec.ClientConfig.URL == nil && e.Spec.ClientConfig.Service == nil {
+		allErrs = append(allErrs, field.Required(
+			specPath.Child("clientConfig"),
+			"either URL or Service must be defined",
+		))
+	}
+	if e.Spec.ClientConfig.URL != nil && e.Spec.ClientConfig.Service != nil {
+		allErrs = append(allErrs, field.Forbidden(
+			specPath.Child("clientConfig"),
+			"only one of URL or Service can be defined",
+		))
+	}
+
+	// Validate URL
+	if e.Spec.ClientConfig.URL != nil {
+		if uri, err := url.ParseRequestURI(*e.Spec.ClientConfig.URL); err != nil {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"must be a valid URL, e.g. https://example.com",
+			))
+		} else if uri.Scheme != "http" && uri.Scheme != "https" {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "url"),
+				*e.Spec.ClientConfig.URL,
+				"'https' and 'http' are the only allowed URL schemes, e.g. https://example.com",
+			))
+		}
+	}
+
+	// Validate Service if defined
+	if e.Spec.ClientConfig.Service != nil {
+		// Validate that the name is not empty and is a Valid RFC1123 name.
+		if e.Spec.ClientConfig.Service.Name == "" {
+			allErrs = append(allErrs, field.Required(
+				specPath.Child("clientConfig", "service", "name"),
+				"must not be empty",
+			))
+		}
+
+		if errs := validation.IsDNS1035Label(e.Spec.ClientConfig.Service.Name); len(errs) != 0 {
+			allErrs = append(allErrs, field.Invalid(
+				specPath.Child("clientConfig", "service", "name"),
+				e.Spec.ClientConfig.Service.Name,
+				"invalid name",
```

Comment:
nit: should we include the validation error in the msg? Otherwise it's hard to know how the name is wrong without looking up the webhook and validation code

Reply:
Same above for the URL validation in l. 143 and below for namespace, .. (please check the whole file)

Reply:
I adjusted the validations using `IsDNS...` to log all provided messages (as done upstream).

Also fixed l. 143

Also fixed one more occurence where the returning value was not used.